ecr/layer_writer_test.go

```diff
@@ -89,17 +90,17 @@ func TestLayerWriter(t *testing.T) {
 	tracker.SetStatus(refKey, docker.Status{})
 
 	lw, err := newLayerWriter(ecrBase, tracker, "refKey", desc)
-	assert.NoError(t, err)
+	require.NoError(t, err)
```

Comment:
I'd like to revert this commit.  There's value in reduced debugging to see all of the failed assertions for a given test, even if it's a bit more information to wade through.

Reply:
This prevents several `nil` pointer panics because values are inspected even when there's an error. I read through the tests as I updated them and tried to use `assert` where there was a coherency check that needed to also happen. Are there additional checks (or reordering) that you'd like to see that could be added to combat the reduced scope?

Reply:
The assertions right here are all simple integer equality assertions that I'd still like to see happen.  You could reorder the `require.NoError` below the integer assertion, or you could `require.NotNil(lw)` at the bottom to avoid the panic on line 98.